Source/Core/Core/HW/EXI/EXIBrawlback.cpp

```diff
@@ -1189,137 +1191,328 @@ void CEXIBrawlback::handleStartMatch(u8* payload)
+  startReplay.otherRandomSeed = swap_endian(startReplay.otherRandomSeed);
+  startReplay.randomSeed = swap_endian(startReplay.randomSeed);
   for (int i = 0; i < startReplay.numPlayers; i++)
   {
-    swap_endian(startReplay.players[i].startPlayer.xPos);
-    swap_endian(startReplay.players[i].startPlayer.yPos);
-    swap_endian(startReplay.players[i].startPlayer.zPos);
+    auto& startPlayer = startReplay.players[i].startPlayer;
+
+    startPlayer.xPos = swap_endian(startPlayer.xPos);
+    startPlayer.yPos = swap_endian(startPlayer.yPos);
+    startPlayer.zPos = swap_endian(startPlayer.zPos);
   }
 }
 
 void CEXIBrawlback::handleStartReplaysStruct(u8* payload)
 {
-  StartReplay startReplay;
-  std::memcpy(&startReplay, payload, sizeof(StartReplay));
-  fixStartReplayEndianness(startReplay);
+  if (SConfig::GetInstance().m_brawlbackSaveReplays)
+  {
+    StartReplay startReplay;
+    std::memcpy(&startReplay, payload, sizeof(StartReplay));
 
-  this->curReplayName = std::string(startReplay.nameBuffer, startReplay.nameBuffer + startReplay.nameSize);
+    this->curReplayName = std::string(startReplay.nameBuffer, startReplay.nameBuffer + startReplay.nameSize);
 
-  auto& start = this->curReplayJson["start"];
-  for (int i = 0; i < startReplay.numPlayers; i++)
-  {
-    auto& player = start["players"][i];
-    auto& position = player["startPlayerPos"];
+    auto& start = this->curReplayJson["start"];
+    for (int i = 0; i < startReplay.numPlayers; i++)
+    {
+      auto& player = start["players"][i];
+      auto& position = player["startPlayerPos"];
 
-    auto replayPlayer = startReplay.players[i];
-    auto replayPosition = replayPlayer.startPlayer;
-	
-    player["ftKind"] = replayPlayer.fighterKind;
+      auto replayPlayer = startReplay.players[i];
+      auto replayPosition = replayPlayer.startPlayer;
 
-    position["x"] = replayPosition.xPos;
-    position["y"] = replayPosition.yPos;
-    position["z"] = replayPosition.zPos;
+      player["ftKind"] = replayPlayer.fighterKind;
+      player["slotID"] = replayPlayer.slotID;
+
+      position["x"] = replayPosition.xPos;
+      position["y"] = replayPosition.yPos;
+      position["z"] = replayPosition.zPos;
+    }
+    start["stage"] = startReplay.stage;
+    start["randomSeed"] = startReplay.randomSeed;
+    start["otherRandomSeed"] = startReplay.otherRandomSeed;
+    start["firstFrame"] = startReplay.firstFrame;
   }
-  start["stage"] = startReplay.stage;
-  start["randomSeed"] = startReplay.randomSeed;
-  start["otherRandomSeed"] = startReplay.otherRandomSeed;
-  start["firstFrame"] = startReplay.firstFrame;
 }
 
 void CEXIBrawlback::fixReplayEndianness(Replay& replay)
 {
-  swap_endian(replay.frameCounter);
-  swap_endian(replay.persistentFrameCounter);
+  replay.frameCounter = swap_endian(replay.frameCounter);
+  replay.persistentFrameCounter = swap_endian(replay.persistentFrameCounter);
   for (int i = 0; i < replay.numItems; i++)
```

Comment:
Both of these loops can be easily rewritten as ranged for loops for readability, e.g.:
```
  for (auto& item : replay.items)
  {
    item.itemId = swap_endian(item.itemId);
    item.itemVariant = swap_endian(item.itemVariant);
  }
```